unison-core/src/Unison/Referent'.hs

```diff
@@ -35,7 +36,7 @@ import Unison.Prelude (Word64)
 -- | When @Ref'@ then @r@ represents a term.
 --
 -- When @Con'@ then @r@ is a type declaration.
-data Referent' r = Ref' r | Con' r ConstructorId ConstructorType
+data Referent' r = Ref' r | Con' (GConstructorReference r) ConstructorType
```

Comment:
What's the deal with the Constructor Type here?
Seems kinda strange that Constructor ID gets folded in but the Constructor Type doesn't; however I don't think we always have the Constructor Type available do we? 

Ideally I think ConstructorType would end up in the GConstructorReference too (or in the TypeReference), but I don't know that it's possible ATM.

Not worth holding up this PR for though 👍🏼